CMakeLists.txt

```diff
@@ -376,6 +379,94 @@ if (CURL_FOUND)
   include_directories("${CURL_INCLUDE_DIRS}")
 endif()
 
+if (ENABLE_ALL OR ENABLE_SFTP)
```

Comment:
Can this be put into a cmake module. We'll slowly move in that direction like libressl did.

Reply:
I think so, I'll try to do that.

Reply:
Done that, added use_bundled_libssh2(), but unlike LibreSSL it must only be called after ZLIB and the SSL implementation (OpenSSL/LibreSSL) has been processed. We're currently calling the function the proper way, but I am not sure if I can programatically protect it from being called at the wrong place.